pkg/domain/stateDiffComponent.go

```diff
@@ -68,11 +68,19 @@ func (diff *ComponentDiffState) String() string {
 	return fmt.Sprintf(
 		"{DistributionNamespace: %q, Version: %q, InstallationState: %q}",
 		diff.DistributionNamespace,
-		diff.Version.String(),
+		diff.getSafeVersionString(),
 		diff.InstallationState,
 	)
 }
 
+func (diff *ComponentDiffState) getSafeVersionString() string {
+	if diff.Version != nil {
+		return diff.Version.String()
+	} else {
+		return "0.0.0"
```

Comment:
Shouldn't this rather be an empty string?

Reply:
Yes :+1: